pennylane/transforms/optimization/optimization_utils.py

```diff
@@ -60,17 +72,47 @@ def _quaternion_product(q1, q2):
     return stack([qw, qx, qy, qz])
 
 
-def _fuse(angles_1, angles_2):
-    """Perform fusion of two angle sets. Separated out so we can do JIT with conditionals."""
-    # Compute the product of the quaternions
-    qw, qx, qy, qz = _quaternion_product(_zyz_to_quat(angles_1), _zyz_to_quat(angles_2))
+def _singular_quat_to_zyz(q, y_arg):
+    """Compute the ZYZ angles for the singular case of qx = qy = 0"""
+    qw, qx, qy, qz = q
+    z1_arg1 = 2 * (qx * qy + qz * qw)
+    z1_arg2 = 1 - 2 * (qx**2 + qz**2)
```

Comment:
Minor: these lines seem to be repeated in `_singular_quat_to_zyz_jax`. Could they not be duplicated?

It would likely be cleaner to put shared code directly into the main function definitions so that in the future code in only one place has to be updated.

Reply:
Good call. I tried this a bit but can have another look at potential optimization. Worst case is that we have to pass around more variables, but that should be fine :)